crates/burn-tensor/src/tensor/api/numeric.rs

```diff
@@ -2030,6 +2030,95 @@ where
         // Assign the original tensor data to the appropriate slice of the padded tensor
         padded_tensor.slice_assign(ranges, self)
     }
+    /// Create a one-hot encoded tensor with configurable `depth`, `on_value`, `off_value`, and `axis` including high-ranked tensors.
+    ///
+    /// # Arguments
+    ///
+    /// * `depth`: The number of classes for the one-hot encoding, which defines the size of the one-hot dimension.
+    /// * `on_value`: The value to assign for active positions (corresponding to indices).
+    /// * `off_value`: The value to assign for inactive positions.
+    /// * `axis`: The axis along which the one-hot dimension is added. Supports negative indexing.
+    ///
+    /// # Returns
+    ///
+    /// A tensor with one additional dimension for the one-hot encoding, where active positions are filled with `on_value` and others with `off_value`.
+    ///
+    /// # Example
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Tensor, Float};
+    /// fn example<B: Backend<FloatElem: From<f32>>>() {
+    ///     let device = B::Device::default();
+    ///     let indices: Tensor<B, 2, Float> = Tensor::from_floats([[0., 2.], [1., -1.]], &device);
+    ///     // One-hot encoding
+    ///     let tensor:Tensor<B, 3, Float> = indices.one_hot_plus(3, 5.0.into(), 0.0.into(), -1);
+    ///     println!("{tensor}");
+    ///     // [[[5.0, 0.0, 0.0],
+    ///     // [0.0, 0.0, 5.0]],
+    ///     // [[0.0, 5.0, 0.0],
+    ///     // [0.0, 0.0, 5.0]]]
+    /// }
+    /// ```
+    pub fn one_hot_plus<K2: Numeric<B>, const D2: usize>(
+        self,
+        depth: usize,
+        on_value: K2::Elem,
+        off_value: K2::Elem,
+        axis: i64,
+    ) -> Tensor<B, D2, K2> {
```

Comment:
Technically speaking, one-hot encoding only has "hot" values encoded as 1 and the rest is 0. I think that's what the majority of use cases require.

For that reason, I think we could split this into `tensor.one_hot(num_classes)` which would have the one-hot dim added to the last dimension and `tensor.one_hot_fill(num_classes, on_value, off_value, axis)` for a more configurable output 🤔

Reply:
Thank you for your suggestions, let me try modify codes, replacing existing `one_hot` functions.